pkg/api/v2/reporter/reporter.go

```diff
@@ -195,20 +203,22 @@ type StatusReporter interface {
 }
 
 type reporter struct {
-	reporterRef  string
-	statusClient resources.StatusClient
-	clients      map[string]ReporterResourceClient
+	reporterRef    string
+	statusClient   resources.StatusClient
+	messagesClient resources.MessagesClient
```

Comment:
since the messages live on the k8s status resource, i think we should have a single `statusClient` that happens to write the status we have today as well as any messages / additional info

Reply:
That does make the code alot less messy too, I can get that fixed up!

Reply:
Fixed